build_tools/doctool/src/com/google/doctool/custom/JavaEmulSummaryDoclet.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2008 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.doctool.custom;
+
+import jdk.javadoc.doclet.Doclet;
+import jdk.javadoc.doclet.DocletEnvironment;
+import jdk.javadoc.doclet.Reporter;
+
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.type.ArrayType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.Diagnostic;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A doclet for using producing EZT output listing the specified classes and
+ * their methods and constructors.
+ */
+public class JavaEmulSummaryDoclet implements Doclet {
+
+    public static final String OPT_EZTFILE = "-eztfile";
+    private static final String JAVADOC_URL = "https://docs.oracle.com/javase/8/docs/api/";
+
+    private Reporter reporter;
+    private String outputFile;
+
+    @Override
+    public boolean run(DocletEnvironment env) {
+        try {
+
+            File outFile = new File(outputFile);
+            outFile.getParentFile().mkdirs();
+            FileWriter fw = new FileWriter(outFile);
+            PrintWriter pw = new PrintWriter(fw, true);
```

Comment:
try-with-resources?

(like these other notes, nearly no value here since this a one-off that won't keep running)